pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -308,26 +308,52 @@ type VitessShardStatus struct {
 	// HasMaster is a condition indicating whether the Vitess topology
 	// reflects a master for this shard.
 	HasMaster corev1.ConditionStatus `json:"hasMaster,omitempty"`
+
 	// HasInitialBackup is a condition indicating whether the initial backup
 	// has been seeded for the shard.
 	HasInitialBackup corev1.ConditionStatus `json:"hasInitialBackup,omitempty"`
-	// MasterAlias is the tablet alias of the master according to the global
-	// shard record. This could be empty either because there is no master,
-	// or because the shard record could not be read. Check the HasMaster
-	// condition whenever the distinction is important.
-	MasterAlias string `json:"masterAlias,omitempty"`
 
 	// Idle is a condition indicating whether the shard can be turned down.
 	// If Idle is True, the shard is not part of the active shard set
 	// (partitioning) for any tablet type in any cell, so it should be safe
 	// to turn down the shard.
 	Idle corev1.ConditionStatus `json:"idle,omitempty"`
 
+	// Conditions is a map of all VitessShard specific conditions we want to set and monitor.
```

Comment:
Let's document the API guarantee that it's ok for other controllers to add conditions here, and we will preserve them, since we are going to rely on that guarantee elsewhere.

Reply:
Good point. Will add that now.

Reply:
Added https://github.com/planetscale/vitess-operator/pull/31/commits/3851de380e22eed9628ca7734f94443f22db3c52.